### ISSUE TITLE
Add welcome message in JS console.

### ISSUE THREAD

**Issue by Carreau** [jupyter/notebook#1664]
A user the open the JS console is likely going to poke around JS, and be a developer.
I think a small message that say, hi, and give them some information as well as encourage the to contribute would be nice !

**Comment by michaelpacer** [comment]
I support this, strongly.

**Comment by Carreau** [comment]
We could do the same on the jupyter website.

**Comment by jinmehta13** [comment]
Hello, I'm new to open source but a big fan of Jupyter Notebooks and hoping to contribute if I can. Any tips to get started on this?

**Comment by Carreau** [comment]
Hi @jinmehta13 welcome !

How experienced are you, and in which domain ?

Depending on how you want to get started I would do the following:

If you want to get started on the website, and add a message in the JS console there, the first step would be to get the website to work on you local machine. The website repo is [there](https://github.com/jupyter/jupyter.github.io), once this work we can see for modifications.

If you like to make that on the notebook, the simplest if first to get this current repo on your computer, and managed to install a development version of the notebook. You should be able to find the documentation [here](http://jupyter-notebook.readthedocs.io/). Once this is done you can start to make local modifications :-) 

One thing that would be great, is while you are doing this: when you find something unclear, hard, or just don't know where to search, take notes and tell us afterward, that will help us to improve the documentation a bit more for the next contributor. 

If you have more questions, feel fre to ask.

**Comment by jinmehta13** [comment]
Hey @Carreau, I finished up my undergrad in Computer Science and Engineering this past May. Still looking for a job but my internship experiences worked with a single page web application and a lot of IPython Notebooks with pandas, NumPy, and matplotlib. Like I mentioned earlier, new to open source but looking to be an involved contributor to a project and this seemed like a good fit given my familiarity with Jupyter from projects.

So it took a little time but I managed to get the website working on my local machine. It'd be nice to work on the notebook too so I'll try to have that setup over the next few days.

Some notes on the setup process for the website (note: I had no prior experience with Ruby, Gems, etc.):

- Installing Ruby and the DevKit was straightforward and I had no issues following the docs

- Installing github-pages and some other dependencies on a Windows 64-bit machine required some work. It might help to mention in the [Readme](https://github.com/jupyter/jupyter.github.io#quick-local-testing) to run `bundle install` to execute the Gemfile already included in the project. At one point I made a Gemfile in the Devkit directory.

- I ran into the following error: `You have already activated json 2.0.3, but your Gemfile requires json 1.8.6. Prepending 'bundle exec' to your command may solve this. (Gem::LoadError)`
I couldn't get past this by specifying `gem 'json', '1.8.6'` in the Gemfile so I have to run `bundle exec jekyll serve --baseurl ''` to serve the project to localhost.

Overall, it was not much trouble to get setup with the website. Could you provide some more specifics or next steps to add this welcome message (what message to output, where some of the core logic that I might be working with is located, etc)? I'll try to familiarize myself with the project structure in the meantime. Thanks!

**Comment by jinmehta13** [comment]
@Carreau so I can add the console message like so:

`<script>console.log('Welcome to Project Jupyter! Explore the various tools available and their corresponding documentation. If you are interested in contributing to the platform, please visit the community resources section at http://jupyter.org/community.html.')</script>`

I added that at the bottom of the "head.html" file but I'm not sure if that is an ideal place to include this code. Also, let me know if the welcome message is alright and if any changes need to be made.

**Comment by Carreau** [comment]
Thanks @jinmehta13 ! That's seem like a good message and a good place. You can make a PR directly on the website repository. 

The message is great !

Thank you for the feedback on the doc as well, we'll see if we can improve the instruction. It's true that few of us have windows machines available and are used to windows, so the instruction are often lacking.

**Comment by jinmehta13** [comment]
@Carreau thanks, PR made with the website repo. I'll get working on the notebook version and I'll provide feedback on that setup process too.

**Comment by Carreau** [comment]
> @Carreau thanks, PR made with the website repo. I'll get working on the notebook version and I'll provide feedback on that setup process too.

Thanks ! It's live !

**Comment by jinmehta13** [comment]
@Carreau, question regarding the console message for the notebook (I got a development version of the notebook up and running by the way): is the message to be displayed in the user's terminal or in the browser console when the notebook app launches?

**Comment by jinmehta13** [comment]
Also, is `notebook/notebook/templates/tree.html` an appropriate place for the message if it's to be displayed in the browser console?

**Comment by Carreau** [comment]
@jinmehta13 I was thinking in the JS console, but we could have something in the terminal as well. It should not be too hard. 

I would try to add th message in one of the `main.js` likely, for the notebook we try to separate cleanly the html from the javascript. 

You can also try to figure out if the developer tools are getting opened, and re-print the message in the JS console if it has been. How does that sounds ?

**Comment by jinmehta13** [comment]
@Carreau, I moved the console message into `..notebook/static/tree/js/main.js`. I looked around between Stack Overflow and the Chrome/Firefox APIs and I don't think there's any sound way to detect if the dev tools are being opened.

Also added the console message into the terminal output between the Control-C exit information and authentication token. That change was made in the `../notebook/notebookapp.py` file.

I'll put in a PR but let me know if you have some ideas on detecting the dev tools usage and I can try to implement that as well.
